Makefile

```diff
@@ -27,4 +27,15 @@ all-tests:
 generate build clean:
 	go $@ $(EXTRAGOARGS)
 
-.PHONY: all generate clean build test unit-tests all-tests
+sandbox-test-fc:
```

Comment:
This rule does two things: Build a container image and run it. It would be best to split this into distinct rules for each task. You might want to use a "stamp" file to avoid having the image creation rule be PHONY. Since it does actually create an artifact, this makes sense (the artifact being the image stored within Docker, rather than a file on the filesystem)

Reply:
> You might want to use a "stamp" file to avoid having the image creation rule be PHONY. Since it does actually create an artifact, this makes sense (the artifact being the image stored within Docker, rather than a file on the filesystem)

What exactly is a stamp file?

Reply:
> What exactly is a stamp file?

It's a technique to avoid having to declare a target PHONY. Make generally assumes that, after the execution of the recipes associated with a target, a file will exist with a name matching the name of the target. Some rules, though, don't create files, but have some other side-effect (e.g. yours, with a side-effect that there's now a new image registered with Docker).

A stamp file is a file that you can create upon successful creation of a rule that would otherwise be PHONY. Its name matches that of the target, so from make's perspective, the target was successfully rebuild and doesn't need to get rebuilt. So, if you consider the following makefile:

```
do_phony_stuff:
	@echo I ran, and I will keep running.
	# Here I might do some time consuming activity

do_stamp_stuff:
	@echo I ran, but only once
	# Here I might do some time consuming activity
	@touch do_stamp_stuff

clean:
	-rm do_stamp_stuff

.PHONY: do_phony_stuff
```

There are two targets that are basically the same. One is phony, one uses a stamp target. If you run the phony target multiple times, it is executed multiple times:
```
$ make do_phony_stuff
I ran, and I will keep running.
# Here I might do some time consuming activity
$ make do_phony_stuff
I ran, and I will keep running.
# Here I might do some time consuming activity
$ make do_phony_stuff
I ran, and I will keep running.
# Here I might do some time consuming activity
$ make do_phony_stuff
I ran, and I will keep running.
# Here I might do some time consuming activity
$ make do_phony_stuff
I ran, and I will keep running.
# Here I might do some time consuming activity
```

Whereas the target that uses a stamp only runs once:
```
$ make do_stamp_stuff
I ran, but only once
# Here I might do some time consuming activity
$ make do_stamp_stuff
make: 'do_stamp_stuff' is up to date.
$ make do_stamp_stuff
make: 'do_stamp_stuff' is up to date.
$ make do_stamp_stuff
make: 'do_stamp_stuff' is up to date.
```

Of course, dependencies can cause it to be rebuilt, as can the non-presense of the stamp file (which is why you should be sure the remove it in the `clean` recipe)

Because Docker caches the result of builds, this isn't super important here. The first call to `docker build` may be time consuming, but subsequent calls will be fairly fast. Still, this is a good practice to be employ in general.